src/MBP_UserImport_BaseSource.php

```diff
@@ -70,6 +70,19 @@ public function __construct($message) {
   abstract protected function setKeys();
   
   /**
+   * Logic to determine if data from row in CSV file can be processed.
+   *
+   * @param array $data
+   *   Values to test.
+   */
+  abstract public function canProcess($data);
+
+  /**
+   * Assign columns specific to the source to common columns expected by the consumer.
+   */
+  abstract public function setter(&$data);
```

Comment:
Bad name for a method, might be confused with http://php.net/manual/en/language.oop5.overloading.php#object.set.